### PR TITLE
[Program: GCI] Fetch user profile picture from GRAVATAR

### LINKED ISSUE #326: Fetch GRAVATAR for User profile using email
**Is your feature request related to a problem? Please describe.**
Currently, the users have an option to set a URL for their profile pictures. Another PR is underway for using AWS for the same, by uploading the images first. However, for a quicker option, it will be good to fetch the profile pictures from GRAVATAR.

Instead of fetching and saving the image, one just needs to grab the URL from GRAVATAR and store it into the profile picture field on the database.

**Describe the solution you'd like**
Fetch GRAVATAR URL using user email and save into database.

### PR DESCRIPTION
**Pull request by daksha-aeer**
### Description
Gravatar URL generated when user clicks on email verification link. It is based on user's email and uses **Identicon style**. Link is stored in "photo_url" field in User database table.

#### Working
- **libgravatar** library installed for URL generation
- Gravatar generated by confirm_registration() function in "dao/user.py". This function is called when GET request is made to **user/confirm_email** route.

Fixes #326

### Type of Change:

- Code

**Code/Quality Assurance Only**
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
1. Email confirmation
task44_1

2. GET /user
task44_3

3. Tests
```sh
python -m unittest discover tests
```
task44_4

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] Update Postman API at /docs folder
- [ ] Update Swagger documentation and the exported file at /docs folder
- [x] Update requirements.txt

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: app/api/dao/user.py [hunk 1]
```diff
@@ -311,6 +311,9 @@ def confirm_registration(token):
         else:
             user.is_email_verified = True
             user.email_verification_date = datetime.now()
+            # gravatar
+            gravatar = Gravatar(user.email)
+            user.photo_url = gravatar.get_image(default = "identicon")
```
Comment (sara-02) [inline app/api/dao/user.py]: Hey, we need to add a gravatar only if `user.photo_url` is empty. We need to check that first.

**Review by sara-02 — CHANGES_REQUESTED** [review]

File: app/api/dao/user.py (hunk 1 above)
Comment (daksha-aeer) [reply app/api/dao/user.py]: Done please review :)

**Review by xprilion — APPROVED** [review]
Looks good to me. Will wait for @sara-02 to check on her requested changes.

**Review by sara-02 — APPROVED** [review]

**Comment by sara-02** [comment]
Just a Pythonic suggestion, while comparing `None` it is prefered to use `if abc is None` rather than `if abc == None`.

**Comment by daksha-aeer** [comment]
> Just a Pythonic suggestion, while comparing `None` it is prefered to use `if abc is None` rather than `if abc == None`.

Done

**Comment by isabelcosta** [comment]
I am closing this because during final evaluations of PRs we decided to proceed with PR #334
Thank you for your contribution during GCI contest 🙌

**Pull request closed without merging**